network/network.go

```diff
@@ -207,3 +218,49 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 	scrapingElapsed := time.Since(scrapingStart)
 	c.logger.Log("level", "info", "message", "collected metrics", "scrapeID", c.scrapeID, "scrapeTime", scrapingElapsed.Seconds())
 }
+
+func (c *Collector) getNeighbours(n int, subsets []v1.EndpointSubset) ([]string, error) {
+	// Find our IP - note: this does not open a connection, due to UDP.
+	conn, err := net.Dial("udp", "8.8.8.8:80")
```

Comment:
Isn't it better to connect to something local? Like IP of the service which is already known at this point?

Reply:
We don't connect to anything, because of UDP.

Reply:
With you.